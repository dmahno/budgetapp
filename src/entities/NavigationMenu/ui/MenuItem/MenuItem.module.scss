.link {
  display: inline-block;
  border-radius: 25px;
  padding: $indent-xs $indent-m;
  text-decoration: none;
  color: var(--navItemLinkColor);
  transition:
    color 0.2s ease-in,
    background-color 0.2s ease-in;
  border: 1px solid transparent;

  &:hover {
    color: var(--navLinkHoverColor);
  }
}

.active {
  color: var(--navLinkActiveColor);
  border-color: var(--navItemLinkActiveColorBorder);
  background-color: var(--navItemLinkActiveColorBG);
  color: var(--navItemLinkActiveColor);
  &:hover {
    color: var(--navLinkHoverActiveColor);
  }
}
