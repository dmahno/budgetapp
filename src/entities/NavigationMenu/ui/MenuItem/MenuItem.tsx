import {NavLink} from 'react-router-dom';

import {IMenuLinksProps} from '../../model/data/menuLinks';
import styles from './MenuItem.module.scss';

export const MenuItem: React.FC<IMenuLinksProps> = ({url, label}) => {
  return (
    <li>
      <NavLink
        className={({isActive}) =>
          isActive ? `${styles.link} ${styles.active}` : styles.link
        }
        to={url}
      >
        {label}
      </NavLink>
    </li>
  );
};
