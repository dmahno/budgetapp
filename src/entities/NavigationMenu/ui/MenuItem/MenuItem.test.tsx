import {describe, it, expect} from 'vitest';

import {render, screen} from 'shared/tests/test-utils';

import {MenuItem} from './MenuItem';

describe('MenuItem Component', () => {
  const mockProps = {
    url: '/test-route',
    label: 'Test Label',
  };

  it('renders the link with correct href and label', () => {
    render(<MenuItem url={mockProps.url} label={mockProps.label} />);

    const linkElement = screen.getByRole('link', {name: mockProps.label});
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute('href', mockProps.url);
  });

  it('applies active class when the route is active', () => {
    render(<MenuItem url={mockProps.url} label={mockProps.label} />, {
      routerProps: {initialEntries: [mockProps.url]},
    });

    const linkElement = screen.getByRole('link', {name: mockProps.label});
    expect(linkElement).toHaveClass('_link_9fe8fa _active_9fe8fa');
  });

  it('applies only link class when the route is inactive', () => {
    render(<MenuItem url={mockProps.url} label={mockProps.label} />, {
      routerProps: {initialEntries: ['/another-route']},
    });

    const linkElement = screen.getByRole('link', {name: mockProps.label});
    expect(linkElement).toHaveClass('_link_9fe8fa');
    expect(linkElement).not.toHaveClass('active');
  });

  it('matches the snapshot', () => {
    const {asFragment} = render(
      <MenuItem url={mockProps.url} label={mockProps.label} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
