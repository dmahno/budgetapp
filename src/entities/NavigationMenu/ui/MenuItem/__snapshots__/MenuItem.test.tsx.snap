// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuItem Component > matches the snapshot 1`] = `
<DocumentFragment>
  <li>
    <a
      class="_link_9fe8fa"
      href="/test-route"
    >
      Test Label
    </a>
  </li>
</DocumentFragment>
`;
